Roles/Core/CustomRoleManager.cs

```diff
@@ -214,6 +222,26 @@ public static bool OnCheckMurder(PlayerControl killer, PlayerControl target)
             return false;
         }
 
+        // Swap controllers if Sheriff shots Dollmasters main body.
+        if (DollMaster.HasEnabled && DollMaster.IsControllingPlayer)
+        {
+            if (killer.Is(CustomRoles.Sheriff) && target == DollMaster.DollMasterTarget)
+            {
+                if (!(DollMaster.DollMasterTarget == null || DollMaster.controllingTarget == null))
+                    if (target == DollMaster.DollMasterTarget || target == DollMaster.controllingTarget)
+                        target = target == DollMaster.controllingTarget ? DollMaster.DollMasterTarget : DollMaster.controllingTarget;
+            }
+        }
```

Comment:
didn't you already do this here? 
![image](https://github.com/0xDrMoe/TownofHost-Enhanced/assets/143632280/aa4c8b22-5d8b-4942-9273-9b9c04e7450f)

Reply:
Yes sheriff has a special case where you have to swap the targets again before killing a possessed player from the dollmaster because of misfire.